src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -291,9 +296,35 @@ public Response getResponse(String userName, String text) {
     // EEK! clean up the API!
     invoke("publishResponse", response);
     invoke("publishResponseText", response);
-    invoke("publishText", response.msg);
+    invoke("publishText", HtmlFilter.stripHtml(response.msg));
     info("to: %s - %s", userName, res);
 
+    // get optional bot surname, if set..
+    Predicates preds = getChat(userName, getCurrentBotName()).predicates;
+    String botname = getCurrentBotName();
+    if (!preds.get("botname").equals("unknown") && !preds.get("botname").isEmpty()) {
+      botname = preds.get("botname");
+    }
+
+    // feed conversation list
+
+    conversationHistory.add("<b>" + userName + " : </b>" + text);
+    conversationHistory.add("<i><b>" + botname + " : </b>" + HtmlFilter.stripHtml(response.msg) + "</i>");

```

Comment:
the conversation history should be a fixed length queue and it should just store the last N response objects.   We shouldn't be mixing html markup with the history here.